CHANGELOG.md

```diff
@@ -15,6 +15,7 @@
 * `RealmConfiguration.Builder.assetFile(Context, String)` has been renamed to `RealmConfiguration.Builder.assetFile(String)`.
 * Object with primary key is now required to define it when the object is created. This means that `Realm.createObject(Class<E>)` and `DynamicRealm.createObject(String)` now throws `RealmException` if they are used to create an object with a primary key field. Use `Realm.createObject(Class<E>, Object)` or `DynamicRealm.createObject(String, Object)` instead.
 * Importing from JSON without the primary key field defined in the JSON object now throws `IllegalArgumentException`.
+* Primary key cannot be changing anymore after the `RealmObject` created.
```

Comment:
I meant that a new (unmanaged) object is created with all fields copied over, like so

``` java
_realm.executeTransaction((realm) -> {
    Person p = realm.where(Person.class).equalTo("id", 42).findFirst():
    // p.setId(43); // this would fail
    Person up = realm.copyFromRealm(p);
    up.setId(43);
    realm.insertOrUpdate(up); // this will work
});
```

So you don't need to _manually copy every field_. But I think we mean the same thing, after all :tongue: